website/static/js/pages/wiki-edit-page.js

```diff
@@ -1,16 +1,153 @@
+                return 'The wiki page name cannot be more than 100 characters.';
+            }
+        },
+        params: function(params) {
+            return JSON.stringify(params);
+        },
+        success: function(response, value) {
+            window.location.href = ctx.urls.base + encodeURIComponent(value) + '/';
+        },
+        error: function(response) {
+            var msg = response.responseJSON.message_long;
+            if (msg) {
+                return msg;
+            } else {
+                // Log unexpected error with Raven
+                Raven.captureMessage('Error in renaming wiki', {
+                    url: ctx.urls.rename,
+                    responseText: response.responseText,
+                    statusText: response.statusText
+                });
+                return 'An unexpected error occurred. Please try again.';
+            }
+        }
+    });
+}
+
+// Apply panels
+$(document).ready(function () {
+    $('*[data-osf-panel]').osfPanel({
+        buttonElement : '.switch',
+        onSize : 'md',
+        'onclick' : function (title, thisbtn, event ) {
+            // this = all the column elements; an array
+            // title = Text of the button
+            // thisbtn = $(this);
+            // thisbtn.hasClass('btn-primary')
 
-ShareJSDoc('.wiki', url, metadata);
\ No newline at end of file
+            $('body').trigger('togglePanel', [
+                title.toLowerCase(),
+                thisbtn.hasClass('btn-primary')
+            ]);
+
+            if (typeof editor !== 'undefined') { ace.edit(editor).resize(); }
+
+        }
+    });
+    $('.openNewWiki').click(function () {
+        $('#newWiki').modal('show');
+    });
+    $('.openDeleteWiki').click(function () {
+        $('#deleteWiki').modal('show');
+    });
+    var panelToggle = $('.panel-toggle'),
+        panelExpand = $('.panel-expand');
+    $('.panel-collapse').on('click', function () {
+        var el = $(this).closest('.panel-toggle');
+        el.children('.wiki-panel.hidden-xs').hide();
+        panelToggle.removeClass('col-sm-3').addClass('col-sm-1');
+        panelExpand.removeClass('col-sm-9').addClass('col-sm-11');
+        el.children('.panel-collapsed').show();
+        $('.wiki-nav').show();
+    });
+    $('.panel-collapsed .wiki-panel-header').on('click', function () {
+        var el = $(this).parent(),
+            toggle = el.closest('.panel-toggle');
+        toggle.children('.wiki-panel').show();
+        el.hide();
+        panelToggle.removeClass('col-sm-1').addClass('col-sm-3');
+        panelExpand.removeClass('col-sm-11').addClass('col-sm-9');
+        $('.wiki-nav').hide();
+    });
+
+    var options = ctx.viewSettings;
+    var menu = 'menu' in options;
+    var edit = 'edit' in options;
+    var compare = 'compare' in options;
+    var view = 'view' in options;
+    // TODO: toggle menu based on url
+    // TODO: Change version select on toggle
+
+    if (edit) {
```

Comment:
This is initial state stuff, which will not be how the rest of the ViewModel works. We could put it in the initialization function but probably not the ViewModel per se.